src/lib/models/event-groups/get-event-in-group.ts

```diff
@@ -1,15 +1,83 @@
-import { isActivityTaskTimedOutEvent } from '$lib/utilities/is-event-type';
+import {
+  isActivityTaskCanceledEvent,
+  isActivityTaskFailedEvent,
+  isActivityTaskTimedOutEvent,
+  isChildWorkflowExecutionCanceledEvent,
+  isChildWorkflowExecutionFailedEvent,
+  isChildWorkflowExecutionTerminatedEvent,
+  isChildWorkflowExecutionTimedOutEvent,
+  isSignalExternalWorkflowExecutionFailedEvent,
+  isTimerCanceledEvent,
+  isWorkflowExecutionCanceledEvent,
+  isWorkflowExecutionFailedEvent,
+  isWorkflowExecutionTerminatedEvent,
+  isWorkflowExecutionTimedOutEvent,
+  isWorkflowTaskTimedOutEvent,
+  isWorkflowTaskFailedEvent,
+} from '$lib/utilities/is-event-type';
 import { isEventGroup } from '$lib/models/event-groups';
 
-export const groupHasActivityTimedOut = (eventGroup: EventGroup): boolean => {
+const eventIsFailureOrTimedOut = (event) => {
+  return (
+    isActivityTaskTimedOutEvent(event) ||
+    isActivityTaskFailedEvent(event) ||
+    isWorkflowExecutionFailedEvent(event) ||
+    isWorkflowExecutionTimedOutEvent(event) ||
+    isWorkflowTaskTimedOutEvent(event) ||
+    isWorkflowTaskFailedEvent(event) ||
+    isChildWorkflowExecutionFailedEvent(event) ||
+    isChildWorkflowExecutionTimedOutEvent(event) ||
+    isSignalExternalWorkflowExecutionFailedEvent(event)
+  );
+};
+
+const groupIsFailureOrTimedOut = (eventGroup: EventGroup): boolean => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const groupEvents = Array.from(eventGroup.events, ([key, value]) => value);
-  return Boolean(groupEvents.find(isActivityTaskTimedOutEvent));
+  return Boolean(groupEvents.find(eventIsFailureOrTimedOut));
 };
 
-export const eventOrGroupHasActivityTimedOut = (
+export const eventOrGroupIsFailureOrTimedOut = (
   event: IterableEvent,
 ): boolean => {
-  if (isEventGroup(event)) return groupHasActivityTimedOut(event);
-  return isActivityTaskTimedOutEvent(event as WorkflowEvent);
+  if (isEventGroup(event)) return groupIsFailureOrTimedOut(event);
+  return eventIsFailureOrTimedOut(event as WorkflowEvent);
+};
+
+const eventIsCanceled = (event) => {
+  return (
+    isActivityTaskCanceledEvent(event) ||
+    isTimerCanceledEvent(event) ||
+    isWorkflowExecutionCanceledEvent(event) ||
+    isChildWorkflowExecutionCanceledEvent(event)
+  );
+};
+
+const groupIsCanceled = (eventGroup: EventGroup): boolean => {
```

Comment:
I wonder if these are better just a methods on the object?

Reply:
Good call, moved them to the EventGroup